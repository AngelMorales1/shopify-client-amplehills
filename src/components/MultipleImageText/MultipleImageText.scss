@import 'settings';
.MultipleImageText {
  &__image-text-container {
    flex-direction: column;
    @media ($medium-up) {
      flex-direction: row;
    }
  }
  &__image {
    max-width: spacing(30);
    max-height: spacing(30);
  }
  &__text {
    text-align: center;
    @media ($medium-up) {
      text-align: start;
    }
  }
}
