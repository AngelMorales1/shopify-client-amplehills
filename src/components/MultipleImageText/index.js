import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import get from 'utils/get';
import styles from './MultipleImageText.scss';
import { Image } from 'components/base';
const MultipleImageText = ({ block, setRef, z }) => {
  const fields = get(block, 'fields', {});
  const imagesAndTexts = get(fields, 'fragmentType.simpleFragments', {});
  const drip = get(fields, 'drip', false);
  const backgroundColor = get(fields, 'backgroundColor', 'white');

  return (
    <div
      style={{ zIndex: z }}
      ref={refBlock => setRef(refBlock)}
      className={cx(
        styles[`MultipleImageText--${backgroundColor}`],
        'py4 px3',
        {
          drip: drip
        }
      )}
    >
      <div className="form-container-width flex flex-column mx-auto">
        {Object.values(imagesAndTexts).map(imageAndText => (
          <div
            key={get(imageAndText, 'uuid', '')}
            className={cx(
              styles['MultipleImageText__image-text-container'],
              'flex justify-between items-center my2'
            )}
          >
            <div className={cx(styles['MultipleImageText__image'], 'm2')}>
              <Image
                className="w100"
                src={get(imageAndText, 'image.data', '')}
              />
            </div>
            <div className="text-container-width w100">
              <p className={cx(styles['MultipleImageText__text'])}>
                {get(imageAndText, 'text', '')}
              </p>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
};
MultipleImageText.propTypes = {
  setRef: PropTypes.func,
  z: PropTypes.number,
  drip: PropTypes.bool,
  block: PropTypes.shape({
    fields: PropTypes.shape({
      contentType: PropTypes.string,
      MultipleImageText: PropTypes.shape({
        simpleFragments: PropTypes.object
      })
    })
  })
};
MultipleImageText.defaultProps = {
  setRef: () => {},
  z: 1,
  drip: false,
  block: {
    fields: {
      contentType: '',
      MultipleImageText: {
        simpleFragments: {}
      }
    }
  }
};
export default MultipleImageText;
