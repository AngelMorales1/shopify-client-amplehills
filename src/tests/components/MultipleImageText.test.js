import React from 'react';
import { shallow } from 'enzyme';
import MultipleImageText from 'components/MultipleImageText';
it('renders without data', () => {
  const component = shallow(<MultipleImageText />);
  expect(component).toMatchSnapshot();
});
it('renders with data', () => {
  const component = shallow(
    <MultipleImageText
      setRef={() => console.log('foo')}
      z={1}
      drip={false}
      block={{
        fields: {
          contentType: 'foo',
          MultipleImageText: {
            simpleFragments: {
              foo: {
                image: { data: 'foo' },
                text: 'foo'
              }
            }
          }
        }
      }}
    />
  );
  expect(component).toMatchSnapshot();
});
