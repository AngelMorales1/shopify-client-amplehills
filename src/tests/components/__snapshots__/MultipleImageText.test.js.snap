// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MultipleImageText
    block={
        Object {
            "fields": Object {
              "MultipleImageText": Object {
                "simpleFragments": Object {
                  "foo": Object {
                    "image": Object {
                      "data": "foo",
                    },
                    "text": "foo",
                  },
                },
              },
              "contentType": "foo",
            },
          }
    }
    drip={false}
    setRef={[Function]}
    z={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="form-container-width flex flex-column mx-auto"
>
        
</div>,
      "className": "py4 px3",
      "style": Object {
        "zIndex": 1,
      },
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "form-container-width flex flex-column mx-auto",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="form-container-width flex flex-column mx-auto"
>
          
</div>,
        "className": "py4 px3",
        "style": Object {
          "zIndex": 1,
        },
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "form-container-width flex flex-column mx-auto",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MultipleImageText
    block={
        Object {
            "fields": Object {
              "MultipleImageText": Object {
                "simpleFragments": Object {},
              },
              "contentType": "",
            },
          }
    }
    drip={false}
    setRef={[Function]}
    z={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="form-container-width flex flex-column mx-auto"
>
        
</div>,
      "className": "py4 px3",
      "style": Object {
        "zIndex": 1,
      },
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "form-container-width flex flex-column mx-auto",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="form-container-width flex flex-column mx-auto"
>
          
</div>,
        "className": "py4 px3",
        "style": Object {
          "zIndex": 1,
        },
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "form-container-width flex flex-column mx-auto",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
